manifests/proxydhcp.pp

```diff
@@ -31,14 +31,20 @@
     $failover = undef
   }
 
+  $dhcp_root_group = (!!$foreman_proxy::dhcp_key_secret or $foreman_proxy::dhcp_manage_acls) ? {
```

Comment:
I don't think I've ever seen `!!` in Puppet and wouldn't know what it does. I think a single `!` is sufficient, but ideally it would have a test. Something like
```ruby
it { is_expected.to contain_class('dhcp').with_dhcp_root_group(expected_group) }
```

The `expected_group` would be either `nil` or `foreman-proxy`, depending on the parameters.